wolfSSL-FIPS-FAQ/src/section02.md

```diff
@@ -0,0 +1,339 @@
+
+## How do I know if I am using the FIPS module?
+
+A: Answer, follow the build instructions in the FIPS User Guide (for cert 3389) or the security policy (for cert 2425). At runtime, call an API that only exists in fips like 'wolfCrypt_GetStatus_fips()'.
+
+NOTE: all FIPS bundles ending in -v2 will come with a directory in wolfssl-root like 'v4.1.0.-SP-and-user-guide' which will contain .pdf copies of the Security Policy and User Guide for that FIPS module version
+
+## Does the Power On Self Test (POST) really have to run every time:
+
+* The app restarts?
+* For each individual process/application?
+* The device power cycles?
+
+A: yes (to all of the above). There is no exception that allows the POST to be skipped. This is a NIST requirement. See https://csrc.gov/CSRC/media/Projects/Cryptographic-Module-Validation-Program/documents/fips140-2/FIPS1402IG.pdf section "9.10 Power-Up Tests for Software Module Libraries" (Excerpt below). "Shall" means "no exceptions"
+
+```
+AS09.08: (Levels 1, 2, 3 and 4) Power-up tests shall be performed by a
+cryptographic module when the module is powered up (after being
+powered off, reset, rebooted, etc.).
+AS09.09: (Levels 1, 2, 3 and 4) The power-up tests shall be initiated
+automatically and shall not require operator intervention.
+TE.09.09.02: The tester shall power-up the module and verify that the
+module performs the power-up self-tests without requiring any operator
+intervention.
+```
+
+### Followup Post Q: What about this feature NO_ATTRIBUTE_CONSTRUCTOR? Can I use that to by-pass the POST by not running it in the constructor?
+
+A: Turning this feature on (assuming it does not cause a compile time error) will immediately make the module no longer FIPS compliant.
+
+1) Once this feature is enabled the operating system itself has to be modified and re-compiled, adding a call to wolfCrypt modules fipsEntry() function any time the OS is starting a new process/application/module that uses the crypto module, or if the OS is booting for the first time following a power cycle.
+
+2) This feature is not for allowing the POST to be skipped or optional, it allows for the POST to be called by the OS after an RNG is initialized or a memory zero (that would erase the POST results) has completed. The call to fipsEntry() SHALL still remain beyond operator control (not optional) per TE.09.09.02 in the above question/answer excerpt
+from https://csrc.nist.gov/CSRC/media/Projects/Cryptographic-Module-Validation-Program/documents/fips140-2/FIPS1402IG.pdf section “9.10 Power-Up Tests for Software Module Libraries”.
```

Comment:
This link does not show up as a link in generated output. Please use Markdown link format, as mentioned above.